[ESIMD][NFC][DOC] Add load/store/prefetch_2d functions, L1/L2 hint co… (#13218)
intel/llvm: sycl/doc/extensions/supported/sycl_ext_intel_esimd/sycl_ext_intel_esimd_functions.md

properties props{cache_hint_L1<cache_hint::uncached>, alignment<4> cache_hint_L1<cache_hint::cached>};
auto vec_b = block_load<float, 16>(f32_ptr + 1, props);
```
### Cache-hint properties
Cache-hint properties (if passed) currently add a restriction on the target-device, it must be a Intel® Arc Series (aka DG2) or Intel® Data Center GPU Max Series (aka PVC).  

> The valid combinations of L1/L2 cache-hints depend on the usage context. It may be one of 4 contexts:

Reviewer: ```suggestion
The valid combinations of L1/L2 cache-hints depend on the usage context.. There are 4 contexts:
```
